webhook/resources/backup/validator.go

```diff
@@ -39,10 +43,22 @@ func (b *backupValidator) Create(request *admission.Request, newObj runtime.Obje
 		return werror.NewInvalidError(fmt.Sprintf("%v is not a *longhorn.Backup", newObj), "")
 	}
 
+	if !util.ValidateName(backup.Name) {
+		return werror.NewInvalidError(fmt.Sprintf("invalid name %v", backup.Name), "")
+	}
+
 	if backup.Spec.BackupMode != longhorn.BackupModeFull &&
 		backup.Spec.BackupMode != longhorn.BackupModeIncremental {
 		return werror.NewInvalidError(fmt.Sprintf("BackupMode %v is not a valid option", backup.Spec.BackupMode), "")
 	}
 
+	if backup.Spec.SnapshotName != "" {
+		if _, err := b.ds.GetSnapshotRO(backup.Spec.SnapshotName); err != nil {
+			return werror.NewInvalidError(errors.Wrapf(err, "failed to get snapshot %v of backup %v", backup.Spec.SnapshotName, backup.Name).Error(), "")
+		}
+	} else {
+		return werror.NewInvalidError(fmt.Sprintf("failed to create the backup %v due to the missing snapshot name", backup.Name), "")
```

Comment:
Is non-empty snpahostName must? What will happen if the field is not set? cc@ChanYiLin@PhanLe1010